src/sd/sd_client.cpp

```
@@ -142,17 +145,24 @@ class SdClientImpl : public transport::ITransportListener {
                                      RequestId(SOMEIP_SD_CLIENT_ID, next_multicast_session_id()),
                                      MessageType::NOTIFICATION,
                                      ReturnCode::E_OK);
-        someip_message.set_payload(sd_message.serialize());
+        auto serialized = sd_message.serialize();
+        if (serialized.empty()) {
+            return false;
+        }
+        someip_message.set_payload(std::move(serialized));
 
         transport::Endpoint const multicast_endpoint(config_.multicast_address, config_.multicast_port);
```

Cppcheck (2.21.0)

[style] The function 'get_timestamp' is never used.

(unusedFunction)